File: contests/leetcode-weekly445/q1.py
# problem:
from typing import *


class Solution:
    def findClosest(self, x: int, y: int, z: int) -> int:
        return 1 if abs(x - z) < abs(y - z) else (0 if abs(x - z) == abs(y - z) else 2)


def main():
    test = Solution()
    print(test.findClosest(2, 7, 4))
    print(test.findClosest(2, 5, 6))


if __name__ == "__main__":
    main()

#
# Created by: ASUS At: 2025/4/13 10:33
#

File: contests/leetcode-weekly503/q1.cpp
// problem: https://leetcode.cn/contest/weekly-contest-503/problems/limit-occurrences-in-sorted-array/
#include <algorithm>
#include <iostream>
#include <map>
#include <ranges>
#include <vector>
#include "../utils/vector_helper.h"


using namespace std;

class Solution {
public:
    vector<int> limitOccurrences(vector<int> &nums, int k) {
        int n = nums.size();
        if (n == 1)
            return nums;
        map<int, int> cnts;
        for (int num: nums) {
            cnts[num]++;
        }
        vector<int> res;
        for (auto &[key, v]: cnts) {
            res.insert(res.end(), min(v, k), key);
        }
        return res;
    }
};

int main(int argc, char *argv[]) {
    Solution test;
    show_vector(test.limitOccurrences(temp_vector({1, 1, 1, 2, 2, 3}), 2));
    show_vector(test.limitOccurrences(temp_vector({1, 2, 3}), 1));
    show_vector(test.limitOccurrences(temp_vector({40, 40, 40}), 1));
}

//
// Created By ASUS At 2026-05-24 10:34
//
